src/OpenStudy/relations.jl

```diff
@@ -1,124 +1,181 @@
 """
-    is_vector_relation(relation::RelationType)
+    is_vector_relation(relation::PMD.RelationType)
 
 Returns true is `relation` is a vector relation.
 """
+function is_vector_relation(relation::PMD.RelationType)
+    return relation == PMD.RELATION_1_TO_N || relation == PMD.RELATION_BACKED
+end
+
+"""
+    _has_relation_attribute(relations::Vector{PMD.Relation}, relation_attribute::String)
+
+    Returns true if there is a relation with attribute 'relation_attribute' in a 'Vector{PMD.Relation}'
+"""
+function _has_relation_attribute(
+    relations::Vector{PMD.Relation},
+    relation_attribute::String,
+)
+    has_relation_attribute = false
+    for relation in relations
+        if relation.attribute == relation_attribute
+            has_relation_attribute = true
+        end
+    end
+    return has_relation_attribute
+end
+
+"""
+    _has_relation_type(relations::Vector{PMD.Relation}, relation_type::PMD.RelationType)
+
+    Returns true if there is a relation with type 'relation_type' in a 'Vector{PMD.Relation}'
+"""
+function _has_relation_type(
+    relations::Vector{PMD.Relation},
+    relation_type::PMD.RelationType,
+)
+    has_relation_type = false
+    for relation in relations
+        if relation.type == relation_type
+            has_relation_type = true
+        end
+    end
+    return has_relation_type
+end
+
+"""
+    _check_relation(data::Data, source::String)
+
+    Returns an error message if there is no relation where collection 'source' is the source element
+"""
+function validate_relation(data::Data, source::String)
+    if !haskey(data.relation_mapper, source)
+        error("Collection $(source) is not the source for any relation in this study")
+    end
+end
+
+"""
+    validate_relation(data::Data, source::String, target::String)
+
+    Returns an error message if there is no relation between collections 'source' and 'target'
+"""
+function validate_relation(data::Data, source::String, target::String)
+    validate_relation(data::Data, source::String)
```

Comment:
The types are not necessary here

Reply:
Are you talking about this dispatch?

https://github.com/psrenergy/PSRClassesInterface.jl/blob/ca66d6d68439d908b6869f832fe0cc6ac43640ba/src/OpenStudy/relations.jl#L84-L89